app/controllers/concerns/authentication.rb

```diff
@@ -42,8 +44,8 @@ def store_location
   private
 
   def current_user
-    Current.user = if session[:current_user_id].present?
-      User.find_by(id: session[:current_user_id])
+    Current.user = if session[:current_user_session_token].present?
+      User.find_by(session_token: session[:current_user_session_token])
```

Comment:
Identify a user by their `session_token` instead of their ID. Since this value will change on each login/logout it's less susceptible to a session reply attack.